src/core/components/radio/RadioGroup.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { Legend } from '@guardian/src-label';
+import { InlineError } from '@guardian/src-user-feedback';
+import {
+	descriptionId,
+	generateSourceId,
+} from '@guardian/src-foundations/accessibility';
+import { fieldset, horizontal, vertical } from './styles';
+import { Props } from '@guardian/src-helpers';
+
+type Orientation = 'vertical' | 'horizontal';
+
+const orientationStyles = {
+	vertical: vertical,
+	horizontal: horizontal,
+};
+
+export interface RadioGroupProps extends Props {
+	id?: string;
+	/**
+	 * Gets passed as the name attribute for each radio button
+	 */
+	name: string;
+	/**
+	 * Appears as a legend at the top of the radio group
+	 */
+	label?: string;
+	/**
+	 * Visually hides the label
+	 */
+	hideLabel?: boolean;
+	/**
+	 * Additional text or component that appears below the label
+	 */
+	supporting?: string | JSX.Element;
+	/**
+	 * The direction in which radio buttons are stacked
+	 */
+	orientation?: Orientation;
+	/**
+	 * If passed, error styling should applies to this radio group. The string appears as an inline error message.
+	 */
+	error?: string;
+	children: JSX.Element | JSX.Element[];
+}
+
+/**
+ * [Storybook](https://guardian.github.io/source/?path=/docs/source-src-radio-radio-group--demo) •
+ * [Design System](https://theguardian.design/2a1e5182b/p/2891dd-radio-button/b/46940d) •
+ * [GitHub](https://github.com/guardian/source/tree/main/src/core/components/radio) •
+ * [NPM](https://www.npmjs.com/package/@guardian/src-radio)
+ *
+ * Radio buttons allow users to make a single selection from a set of options.
+ *
+ * The following themes are supported: `default`, `brand`
+ */
+export const RadioGroup = ({
+	id,
+	name,
+	label,
+	hideLabel = false,
+	supporting,
+	orientation = 'vertical',
+	error,
+	cssOverrides,
+	children,
+	...props
+}: RadioGroupProps) => {
+	const groupId = id || generateSourceId();
+	const legend = label ? (
+		<Legend text={label} supporting={supporting} hideLabel={hideLabel} />
+	) : (
+		''
+	);
+	const message = error && (
+		<InlineError id={descriptionId(groupId)}>{error}</InlineError>
+	);
+
+	return (
+		<fieldset
+			aria-invalid={!!error}
```

Comment:
Any chance we can add a role="radiogroup"` while we're here and fix #838

Reply:
I get an eslint error when I do that 😢 

> Non-interactive elements should not be assigned interactive
> [roles.eslintjsx-a11y/no-noninteractive-element-to-interactive-role](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-to-interactive-role.md)

Reply:
This ought to have been fixed in jsx-eslint/eslint-plugin-jsx-a11y#746 so I'm not sure what's happening there. Let's leave it for now, I'll update #838